Utilities/build-script-helper.py

```diff
@@ -8,230 +8,306 @@
 import subprocess
 import sys
 import json
+from typing import List, Optional, Dict
 
-def error(message):
-  print("--- %s: error: %s" % (os.path.basename(sys.argv[0]), message), file=sys.stderr)
-  sys.stderr.flush()
-  raise SystemExit(1)
-
-def get_build_target(swift_exec, args):
-  """Returns the target-triple of the current machine or for cross-compilation."""
-  try:
-    command = [swift_exec, '-print-target-info']
-    target_info_json = subprocess.check_output(command,
-                         stderr=subprocess.PIPE, universal_newlines=True).strip()
-    args.target_info = json.loads(target_info_json)
-    if platform.system() == 'Darwin':
-      return args.target_info["target"]["unversionedTriple"]
-    return args.target_info["target"]["triple"]
-  except Exception as e:
-    # Temporary fallback for Darwin.
-    if platform.system() == 'Darwin':
-      return 'x86_64-apple-macosx'
+
+# -----------------------------------------------------------------------------
+# General utilities
+
+
+def fatal_error(message):
+    print(message, file=sys.stderr)
+    raise SystemExit(1)
+
+
+def escapeCmdArg(arg: str) -> str:
+    if '"' in arg or " " in arg:
+        return '"%s"' % arg.replace('"', '\\"')
     else:
-      error(str(e))
-
-def swiftpm(action, swift_exec, swiftpm_args, env=None):
-  cmd = [swift_exec, action] + swiftpm_args
-  print(' '.join(cmd))
-  subprocess.check_call(cmd, env=env)
-
-def swiftpm_bin_path(swift_exec, swiftpm_args, env=None):
-  swiftpm_args = list(filter(lambda arg: arg != '-v' and arg != '--verbose', swiftpm_args))
-  cmd = [swift_exec, 'build', '--show-bin-path'] + swiftpm_args
-  print(' '.join(cmd))
-  return subprocess.check_output(cmd, env=env, universal_newlines=True).strip()
-
-def get_swiftpm_options(swift_exec, args):
-  swiftpm_args = [
-    '--package-path', args.package_path,
-    '--build-path', args.build_path,
-    '--configuration', args.configuration,
-  ]
-
-  if args.verbose:
-    swiftpm_args += ['--verbose']
-
-  if args.sanitize:
-    for san in args.sanitize:
-      swiftpm_args += ['--sanitize=%s' % san]
-
-  if platform.system() == 'Darwin':
-    swiftpm_args += [
-      # Relative library rpath for swift; will only be used when /usr/lib/swift
-      # is not available.
-      '-Xlinker', '-rpath', '-Xlinker', '@executable_path/../lib/swift/macosx',
-    ]
-  else:
-    swiftpm_args += [
-      # Dispatch headers
-      '-Xcxx', '-I', '-Xcxx',
-      os.path.join(args.toolchain, 'lib', 'swift'),
-      # For <Block.h>
-      '-Xcxx', '-I', '-Xcxx',
-      os.path.join(args.toolchain, 'lib', 'swift', 'Block'),
+        return arg
+
+
+def check_call(cmd: List[str], cwd: Optional[str] = None, env: Dict[str, str] = dict(os.environ), verbose: bool = False):
```

Comment:
Is there a reason you're defaulting to `os.environ` instead of `None` to match the underlying `subprocess.check_call`?

Reply:
No, no particular reason. I changed it now.